src/main/java/de/ustutt/iaas/cc/core/QueueTextProcessor.java

```diff
@@ -21,24 +42,83 @@
  */
 public class QueueTextProcessor implements ITextProcessor {
 
-	private final static Logger logger = LoggerFactory.getLogger(QueueTextProcessor.class);
+	final Logger logger = LoggerFactory.getLogger(QueueTextProcessor.class);
+
+	private QueueConnectionFactory conFactory;
+	private QueueConnection connection;
+	private QueueSession session;
+	private Queue textProcessorRequests;
+	private Queue textProcessorResponses;
+	private QueueSender textProcessorRequestSender;
+	private QueueReceiver textProcessorResponseReceiver;
+	private AsyncReceiver asyncReceiver;
+	
+	private ConcurrentHashMap<UUID, CompletableFuture<String>> responseMap ;
+
+	private final String REQUEST_QUEUE_NAME = "text-processor-requests";
+	private final String RESPONSE_QUEUE_NAME = "text-processor-responses";
+	private final String AWS_CREDENTIALS = "aws.properties";
 
 	public QueueTextProcessor(TextProcessorConfiguration conf) {
-		super();
+		super();		
 		logger.debug("Initializing QueueTextProcessor.");
 		
-		// TODO initialize JMS stuff, connect to queues, ...
+		this.responseMap = new ConcurrentHashMap<UUID, CompletableFuture<String>>();
 		
+		try {
+			conFactory = SQSConnectionFactory.builder().withRegion(Region.getRegion(Regions.US_WEST_2))
+					.withAWSCredentialsProvider(new PropertiesFileCredentialsProvider(AWS_CREDENTIALS)).build();
+
+			connection = conFactory.createQueueConnection();
+			session = connection.createQueueSession(false, Session.AUTO_ACKNOWLEDGE);
+			textProcessorRequests = session.createQueue(REQUEST_QUEUE_NAME);
+			textProcessorResponses = session.createQueue(RESPONSE_QUEUE_NAME);
+			textProcessorRequestSender = session.createSender(textProcessorRequests);
+			textProcessorResponseReceiver = session.createReceiver(textProcessorResponses);
+			asyncReceiver = new AsyncReceiver(this);
+			textProcessorResponseReceiver.setMessageListener(asyncReceiver);
+			connection.setExceptionListener(asyncReceiver);
+			connection.start();
+			
+		} catch (JMSException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+
 	}
 
 	@Override
 	public String process(String text) {
-		// text processing placeholder, to be replaced by your solution :-)
-		String processedText = "[processed by incomplete QueueTextProcessor] - " + text;
-
-		// TODO send text to request queue, receive processed text from response queue, return
+		String processedText = "";
+		try {
+			TextMessage message = session.createTextMessage(text);
+			message.setStringProperty("app", "My first Notebook App");
+			UUID id = UUID.randomUUID();
+			message.setStringProperty("cid", id.toString());
+			textProcessorRequestSender.send(message);
+			
+			//Async Receive Call
+			CompletableFuture<String> futureResponse = new CompletableFuture<String>();
+			responseMap.put(id, futureResponse);

```

Comment:
This should be done before sending the message to ensure that the entry is available in the map whenever the response arrives.